htdocs/core/lib/functions2.lib.php

```diff
@@ -1183,7 +1183,7 @@ function check_value($mask,$value)
 
     $maskraz=-1;
     $maskoffset=0;
-    if (dol_strlen($maskcounter) < 3) return 'ErrorCounterMustHaveMoreThan3Digits';
+    if (dol_strlen($maskcounter) < 3) return -1;
```

Comment:
If we return -1 for all kind of error, we loos an important important on the reason of error into a component that is sensitive.

Reply:
the output of this funciton is always interpreted as an integer by their caller, returning a string instead of an int makes the comparison with 0 fail and always consider this function successful.